docs/user-event/intro.mdx

```diff
@@ -11,36 +11,35 @@ events that would happen if the interaction took place in a browser.
 
 These docs describe `user-event@14`. We recommend updating your projects to this
 version, as it includes important bug fixes and new features. You can find the
-docs for `user-event@13.5.0` [here](../ecosystem-user-event.mdx).
+docs for `user-event@13.5.0` [here](../ecosystem-user-event.mdx), and the changelog for the release [here](https://github.com/testing-library/user-event/releases/tag/v14.0.0).
 
 :::
 
 While most examples with `user-event` are for `React`, the library can be used
 with any framework as long as there is a DOM.
 
-## Difference to `fireEvent`
+## Differences with `fireEvent`
 
 The built-in [`fireEvent`](dom-testing-library/api-events.mdx#fireevent) is a
-utility to easily dispatch events. It dispatches exactly the events you tell it
-to and just those - even if those exact events never had been dispatched in a
-real interaction in a browser.
+utility for easily dispatching events. It dispatches exactly the events you tell it
+to and just those - even if those exact events would not have been dispatched in a
```

Comment:
This is wrong grammar according to [Cambridge Dictionary](https://dictionary.cambridge.org/de/grammatik/britisch-grammatik/conditionals-if).

Reply:
Thanks for the comment @ph-fritsche! Can you be a little more specific about what is wrong? I believe this is correct under "Imagined conditions: The third conditional", in the same entry you referenced. 

I'll admit others may beat me at finding grammar rules, but I can say that, as a native English speaker and former copy editor, this sounds far more correct and native to me than the original wording, which I think is both confusing and incorrect.

I did make an edit to try to bring the whole sentence a little closer to grammatical correctness. (I was going, in the first place, for greater clarity rather than technical grammar, but I agree that both are valuable.)

Reply:
"Imagined conditions: The third conditional" lists the past perfect (3rd form) as the correct conjugation for an imagined condition that would never happen: "even if those had been"
A "would" would be part of the main clause not the conditional. But as the main clause precedes the conditional and is not imaginary itself, thus not subjunctive, I think it's correct to omit it here.

I'm a native German speaker, the use of "would" in the conditional *feels* right to me as English lacks a Konjunktiv and the irrealis mood is only inferred by the time relation between the conditional and the main clause. But I was told it is only used for courtesy in spoken English, but is considered incorrect in formal writing.

> "He is on his way and will probably forget this important thing."
> Courtesy/Spoken: "And even if you would have written him a note, he would still forget it."
> Written: "And even if someone had written him a note, he would still forget it."

Reply:
@ph-fritsche , thanks for clarifying! 

The technical grammar of English is indeed tricky, and is made trickier by the fact that, as native speakers, we tend not to study it formally, so even if we know something is correct/incorrect, it's often difficult for us to point to the grammar rule that says so. 

In reference to the use of "would" in the conditional - in the example you gave

>  "And even if you would have written him a note, he would still forget it."

We'd actually say, or write - 

> "Even if you had written him a note, he would still have forgotten it." 

(Informal, "Even if you'd written him a note, he'd still have forgotten it.")

Or, if it hasn't happend yet: 

> "Even if you write him a note, he will still forget."
 
Or speaking in general: 

> "Even if you write him a note, he still forgets." OR "Even if you write him a note, he will still forget."

For the sentence in question here:

**Previous Form:** 

> It dispatches exactly the events you tell it to and just those - even if those exact events never had been dispatched in a
real interaction in a browser.

This is incorrect. "Never had been" refers to something concrete that actually happened; this is an imagined instance, grammatically. Further, it would be "had never been" (though that's not correct here either).

**Proposed Form:**
 
> It dispatches exactly, and only, the events you tell it to - even if those exact events would not have been dispatched in a real browser interaction.

I believe this is the correct form for describing a hypothetical conditional. The "would" is necessary; if you drop it and just write "had not been," it incorrectly refers to a concrete, non-hypothetical event. If you replace it with "will", it incorrectly refers to a concrete future. Both forms add confusion.

Do you have a specific alternative form you believe would be more correct? I'm not opposed to changing the wording, if we can find something that feels correct to both of us. 

Also, if it would be helpful, I can solicit feedback from other engineers at my workplace on this PR. There are plenty of native English speakers there, and even a few former teachers. We have writers on staff, too, come to think of it. 

Let me know how you'd like to proceed.

Reply:
>This is incorrect. "Never had been" refers to something concrete that actually happened; this is an imagined instance, grammatically.

"even if it never had been" is already *irrealis* as per being a conditional referring to the past.
According to Oxford Dictionary it isn't ambiguous because the main clause is in present tense, so a non-hypothetical conditional would need to be in present or present perfect.
In a *conditional III* the correct phrasing would be:
> Even if the events never had been dispatched in a browser, the utility would dispatch exactly those.

I guess the confusion is caused by the main clause omitting the "would". Which in the case of inversion *feels* correct to me, as we describe that it factually does dispatch those events, but I don't have any dictionary reference to back this up.

Maybe this discussion boils down to differences between US and Oxford English.

> However, some varieties of English regularly use would (contracted to 'd) and would have ('d have) in counterfactual condition clauses, although this is often considered non-standard:
>
>      If you'd leave (standard: you left) now, you'd be on time.
>      If you'd have told (standard: you had told) me, we could've done something about it.
>
>Such use of would is widespread especially in spoken [American English](https://en.wikipedia.org/wiki/American_English) in all sectors of society. It is not usually found in more formal writing; however some sources describe it as acceptable US English, no longer labeling it colloquial.[[8]](https://en.wikipedia.org/wiki/English_conditional_sentences#cite_note-8)[[9]](https://en.wikipedia.org/wiki/English_conditional_sentences#cite_note-9)

As mentioned above, using the would in the conditional here *feels* better to me too. But this is something I've been scolded for too much. ;)
If we can ask some native English speakers who also have a professional background on linguistics, let's do it!
I'm more than happy about a workaround for the lack of Konjunktiv in the English language. \*wavingmyfistatformerteachers\* ;D

Reply:
@ph-fritsche 
> As mentioned above, using the would in the conditional here feels better to me too. But this is something I've been scolded for too much. ;)

Haha, okay! I'll find someone with a linguistic background and ask them to comment!